dynamiqs/integrators/apis/dsmesolve.py

```diff
@@ -410,3 +410,6 @@ def _check_dsmesolve_args(
     if exp_ops is not None:
         for i, E in enumerate(exp_ops):
             check_shape(E, f'exp_ops[{i}]', '(n, n)')
+
+    # === check layout
+    check_layout(rho0, 'rho0')
```

Comment:
Let's move this just below the shape check of `rho0` (just above), and rename the comment
```diff
- # === check rho0 shape
+ # === check rho0
```

Reply:
Also in `mesolve.py`.